### PR TITLE
fix: tmp outage (#1196)

### PR DESCRIPTION
**Pull request by CoolnsX**
fix: remove problematic provider (rab)
fix: accidental skipping of links

# Pull Request Template

## Type of change

- [x] Bug fix
- [ ] Feature
- [ ] Documentation update

## Description

*ramble here*

## Checklist

- [ ] any anime playing
- [ ] bumped version
---
- [ ] next, prev and replay work
- [ ] `-c` history and continue work
- [ ] `-d` downloads work
- [ ] `-s` syncplay works
- [ ] `-q` quality works
- [ ] `-v` vlc works
- [ ] `-e` select episode works
- [ ] `-S` select index works
- [ ] `-r` range selection works
- [ ] `--dub` and regular (sub) mode both work
- [ ] all providers return links (not necessarily on a single anime, use debug mode to confirm)
---
- [ ] `-h` help info is up to date
- [ ] Readme is up to date
- [ ] Man page is up to date

## Additional Testcases

- The safe bet: One Piece
- Episode 0: Saenai Heroine no Sodatekata ♭
- Unicode: Saenai Heroine no Sodatekata ♭
- Non-whole episodes: Tensei shitara slime datta ken (ep. 24.5, ep. 24.9)

### CODE REVIEW

**Review by port19x — CHANGES_REQUESTED** [review]
Generally looks good, I'll give this a few hours for testing by various users tho

**Comment by CoolnsX** [comment]
> Generally looks good, I'll give this a few hours for testing by various users tho

user_1(my dumb personality) confirms it's working
user_2(my autistic personality) confirms it's working
user_3(my weeb personality) confirms it's working

File: ani-cli [hunk 1]
```diff
@@ -175,15 +174,12 @@ get_episode_url() {
...
     episode_embed_gql="query (\$showId: String!, \$translationType: VaildTranslationTypeEnumType!, \$episodeString: String!) {    episode(        showId: \$showId        translationType: \$translationType        episodeString: \$episodeString    ) {        episodeString sourceUrls    }}"
 
-    resp=$(curl -e "$allanime_base" -s --cipher "AES256-SHA256" -G "${allanime_api}/api" --data-urlencode "variables={\"showId\":\"$id\",\"translationType\":\"$mode\",\"episodeString\":\"$ep_no\"}" --data-urlencode "query=$episode_embed_gql" -A "$agent" | tr '{}' '\n' | sed 's|\\u002F|\/|g;s|\\||g' | sed -nE 's|.*sourceUrl":"##([^"]*)".*sourceName":"([^"]*)".*|\2 :\1|p')
+    resp=$(curl -e "$allanime_base" -s --cipher "AES256-SHA256" -G "${allanime_api}/api" --data-urlencode "variables={\"showId\":\"$id\",\"translationType\":\"$mode\",\"episodeString\":\"$ep_no\"}" --data-urlencode "query=$episode_embed_gql" -A "$agent" | tr '{}' '\n' | sed 's|\\u002F|\/|g;s|\\||g' | sed -nE 's|.*sourceUrl":"--([^"]*)".*sourceName":"([^"]*)".*|\2 :\1|p')
     # generate links into sequential files
-    provider=1
-    i=0
     cache_dir="$(mktemp -d)"
-    providers="1 2 3 4 5 6"
+    providers="$(seq 1 5)"
```
[the remark this answers is not in the document]
Comment (71zenith) [reply ani-cli]: done

**Comment by SwedishBarbarossa** [comment]
This works for me as well.

**Review by rahulkarmakar2023cse — APPROVED** [review]

**Review by port19x — APPROVED** [review]

**Pull request merged**